packages/adapter-node/src/server.js

```diff
@@ -64,8 +66,21 @@ export function createServer({ render }) {
 
 			if (rendered) {
 				res.writeHead(rendered.status, rendered.headers);
-				if (rendered.body) res.write(rendered.body);
-				res.end();
+				if (
+					rendered.body &&
+					typeof rendered.body === 'object' &&
+					typeof rendered.body[Symbol.asyncIterator] === 'function'
+				) {
+					const flush = rendered.headers && compressible(rendered.headers['content-type']) ? res.flush : null;
+					const data = Readable.from(rendered.body);
+					data.on('error', () => res.end());
+					if (flush) {
+						data.on('data', () => flush());
+					}
+					data.pipe(res);
```

Comment:
I don't think `pipe` closes the readable stream, does it? We might need `res.on('close', () => data.destroy());`.